client/components/tinymce/plugins/contact-form/plugin.js

```diff
@@ -23,26 +26,38 @@ const contactForm = editor => {
 	} );
 
 	editor.on( 'remove', () => {
-		ReactDom.unmountComponentAtNode( node );
+		unmountComponentAtNode( node );
 		node.parentNode.removeChild( node );
 		node = null;
 	} );
 
-	editor.addCommand( 'WP_ContactForm', () => {
-		function onClose() {
-			editor.focus();
-			renderModal( 'hide' );
-		};
+	editor.addCommand( 'wpcomContactForm', content => {
+		if ( content ) {
+			store.dispatch( loadForm( deserialize( content ) ) );
+		}
 
 		function renderModal( visibility = 'show' ) {
-			ReactDom.render(
-				React.createElement( ContactFormDialog, {
-					showDialog: visibility === 'show',
-					onClose,
-					onInsertMedia( markup ) {
-						editor.execCommand( 'mceInsertContent', false, markup );
-					}
-				} ),
+			render(
+				createElement( Provider, { store },
+					createElement( ContactFormDialog, {
+						showDialog: visibility === 'show',
+						onAdd() {
```

Comment:
I agree the usage here is a bit unusual. Did you have problems connecting this component? Maybe take a look at `client/components/data/query-posts/index.jsx` as an example.